core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -2541,3 +2542,18 @@ top = topBy $ flip compare
 {-# INLINE bottom #-}
 bottom :: (MonadIO m, Storable a, Ord a) => Int -> Fold m a (MA.Array a)
 bottom = topBy compare
+
+{-# INLINE nub #-}
+nub :: (MonadIO m, Ord a) => Fold m a (Maybe a)
```

Comment:
There is no documentation, we should try to write at least a minimal doctest example if possible, so that the usage is clear.